src/httprpc.cpp

```diff
@@ -294,21 +296,26 @@ static bool InitRPCAuthentication()
 {
     if (gArgs.GetArg("-rpcpassword", "") == "")
     {
-        LogInfo("Using random cookie authentication.\n");
-
         std::optional<fs::perms> cookie_perms{std::nullopt};
         auto cookie_perms_arg{gArgs.GetArg("-rpccookieperms")};
         if (cookie_perms_arg) {
             auto perm_opt = InterpretPermString(*cookie_perms_arg);
             if (!perm_opt) {
-                LogInfo("Invalid -rpccookieperms=%s; must be one of 'owner', 'group', or 'all'.\n", *cookie_perms_arg);
+                LogError("Invalid -rpccookieperms=%s; must be one of 'owner', 'group', or 'all'.\n", *cookie_perms_arg);
                 return false;
             }
             cookie_perms = *perm_opt;
         }
 
-        if (!GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)) {
-            return false;
+        if (std::optional<bool> result{GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)}) {
```

Comment:
You gotta' love these 3-state-booleans :)

nit: To reduce the noise, consider simplified to either
```suggestion
        if (std::optional result{GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)}) {
```
or
```suggestion
        if (auto result{GenerateAuthCookie(&strRPCUserColonPass, cookie_perms)}) {
```

Reply:
I prefer documenting the type.

Reply:
it's quite noisy in this case, but don't have string feelings about it :)